controllers/productController.js

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import fs from 'fs';
+import util from 'util';
+import {validate} from '../helpers/validator';
+import {merge} from 'lodash';
+const filePath = path.join(__dirname, '../mock/products.json');
+const readFile = util.promisify(fs.readFile);
+
+const schema = {
+    id: {
+        type: 'number',
+        required: true
+    },
+    name: {
+        type: 'string',
+        required: true
+    },
+    brand: {
+        type: 'string',
+        required: true
+    },
+    price: {
+        type: 'number',
+        required: true
+    },
+    options: {
+        type: 'object',
+        properties: {
+            color: {
+                type: 'string',
+                required: true
+            },
+            size: {
+                type: 'string',
+                required: false
+            }
+        }
+    }
+};
+
+export function getAllProducts(req, res, next) {
+    readFile(filePath).then( (file) => {
+        let data = file.toString();
+        res.writeHead(200, {
+            'Content-Type': 'application/json; charset=utf-8'
+        });
+        res.end(data);
+        console.log('Time:', Date.now());
+        next();
```

Comment:
There's no need to call controller, as all middlewares should be executed before. Controller is a final point where we form and send the response.